src/sources/commondatamodel.jl

```diff
@@ -339,6 +343,48 @@ function _cdmlookup(
     end
     return _cdmlookup(D, index, order, span, sampling, metadata, crs, mappedcrs)
 end
+
+function _get_crs_by_hook_or_crook(ds::AbstractDataset, var, attr)
+    # Get CRS by hook or by crook - start at the variable and work our way up
+    crs = nokw
+    if haskey(attr, "grid_mapping")
+        if haskey(ds, attr["grid_mapping"])
+            crs = _crs_from_cf_attr(CDM.attribs(ds[attr["grid_mapping"]]))
+            !isnokw(crs) && return crs
+        else
+            crs = _crs_from_cf_attr(attr)
+            !isnokw(crs) && return crs
+        end
+    elseif haskey(CDM.attribs(var), "grid_mapping") 
+        var_attr = CDM.attribs(var)
+        if haskey(ds, var_attr["grid_mapping"])
+            crs = _crs_from_cf_attr(CDM.attribs(ds[var_attr["grid_mapping"]]))
+            !isnokw(crs) && return crs
+        else
+            crs = _crs_from_cf_attr(var_attr)
+            !isnokw(crs) && return crs
+        end
+    else# nothing locally...check global attributes as a last resort
+        return _crs_from_cf_attr(attr)
+    end
+
+end
+
+function _crs_from_cf_attr(attr)
+    # we can't parse CF crs fully, BUT
+    # many CF implementations will include either `crs_wkt`, `spatial_epsg`, or `proj4string`
+    # which we do know the types of
+    if haskey(attr, "crs_wkt")
+        return ESRIWellKnownText(GeoFormatTypes.CRS(), attr["crs_wkt"])
```

Comment:
An reason this is `ESRIWellKnown` text and not the normal one?

Reply:
Not particularly, no
```suggestion
        return WellKnownText2(GeoFormatTypes.CRS(), attr["crs_wkt"])